resources/snakefiles/mapping.smk

```diff
@@ -119,18 +119,19 @@ rule sort_index_bam:
     input:
         aln="output/mapping/{mapper}/mapped_reads/{read_sample}_Mapped_To_{contig_sample}.bam"
     output:
-        bam=temp("output/mapping/{mapper}/sorted_bams/{read_sample}_Mapped_To_{contig_sample}.sorted.bam"),
-        bai=temp("output/mapping/{mapper}/sorted_bams/{read_sample}_Mapped_To_{contig_sample}.sorted.bam.bai")
+        bam=temp("output/mapping/{mapper}/sorted_bams/{read_sample}_Mapped_To_{contig_sample}.bam"),
+        index=temp("output/mapping/{mapper}/sorted_bams/{read_sample}_Mapped_To_{contig_sample}.bam.bai")
     conda:
         "../env/mapping.yaml"
     threads:
         config['threads']['sort_bam']
     benchmark:
-        "output/benchmarks/sort_bam/{mapper}/{read_sample}_Mapped_To_{contig_sample}.sorted.txt"
+        "output/benchmarks/mapping/{mapper}/sort_index_bam/{read_sample}_Mapped_To_{contig_sample}.txt"
     log:
-        "output/logs/sort_bam/{mapper}/{read_sample}_Mapped_To_{contig_sample}.sorted.log"
+        "output/logs/mapping/{mapper}/sort_index_bam/sort_index_bam/{read_sample}_Mapped_To_{contig_sample}.log"
     shell:
         """
         samtools sort -o {output.bam} -@ {threads} {input.aln} 2> {log}
         samtools index -b -@ {threads} {output.bam} 2>> {log}
+        touch {output.index}
```

Comment:
You shouldn't need this touch either, because the index .bai is actually created by samtools index, right? 

A better option for validating output-less job execution is using the `touch()` function in the `output` declaration, and use a filename that makes it clear that it is a status signifier (like `samtools_sort_{whatever wildcards}.done`). Then you don't have to explicitly run a `touch` statement in your job script, snakemake will handle it for you on successful execution or the rule.

Reply:
Ok, done. What is the situation where this becomes an issue, though? As I mentioned, I've seen this in lots of similar workflows, and I though you said that people do it for this exact purpose.

Reply:
The only reason I've seen a `touch {output}` in a workflow is when the job is either not producing an output file, or not producing a *known* output file (e.g. jobs producing variable numbers of outputs). You would never want to touch an already existing file -- if someone has that in a workflow, I think it's in error. The `touch` function within snakemake may not be known to everyone putting workflows up, and also hasn't been around forever.